src/components/Datepicker/sgds-datepicker.ts

```diff
@@ -0,0 +1,307 @@
+import { html } from "lit";
+import { customElement, property } from "lit/decorators.js";
+import { ref } from "lit/directives/ref.js";
+import { DatepickerDropdownElement } from "../../base/datepicker-dropdown-element";
+import "./sgds-datepicker-calendar";
+import "./sgds-datepicker-header";
+import styles from "./sgds-datepicker.scss";
+import { live } from "lit/directives/live.js";
+
+export type DateFormat = "MM/DD/YYYY" | "DD/MM/YYYY" | "YYYY/MM/DD";
+
+/**
+ * @summary The `DatePicker` Component is built using `Dropdown`, `Input` and `Button` components. By default, the Calendar points to current date and input has no value. The input is a read-only and users can only pick dates using the Calendar.
+ *
+ * @event sgds-change-date - Emitted when the state of datepicker's input changes during first load, close button reset click & date click. Date values can be accessed via event.target.value
+ *
+ * @cssproperty --datepicker-theme-color - Datepicker's overall theme color
+ * @cssproperty --datepicker-hover-background-color - Datepicker's calendar menu hover color
+ * @cssproperty --datepicker-background-color - Datepicker's menu background color
+ * @cssproperty --datepicker-closebutton-background-color - Datepicker's close button background color
+ * @cssproperty --datepicker-closebutton-hover-background-color - Datepicker's close button hover background color
+ * @cssproperty --datepicker-closebutton-color - Datepicker's close button color
+ */
+
+@customElement("sgds-datepicker")
+export class SgdsDatepicker extends DatepickerDropdownElement {
+  static styles = [DatepickerDropdownElement.styles, styles];
+
+  /** When true, adds required attribute to input element */
+  @property({ type: Boolean, reflect: true }) required = false;
+
+  /** When true, adds disabled attribute to input and button element */
+  @property({ type: Boolean, reflect: true }) disabled = false;
+
+  /** When true, adds no flip even when placement does not fit */
+  @property({ type: Boolean, reflect: true, state: false }) noFlip = false;
+
+  /** @internal */
+  @property({ type: String })
+  view: string;
+
+  /** @internal */
+  @property({ type: Array }) selectedDateRange: Date[] = [];
+
+  /** @internal */
+  @property({ attribute: false }) displayDate: Date = new Date();
+
+  /** @internal */
+  @property({ attribute: false }) displayDateInput: Date;
+
+  /** The initial value range of DatePicker on first load for single & range mode. eg.'["2023-06-22", "2023-06-11"]' */
+  @property({ type: Array }) initialValue: Date[];
```

Comment:
is initialValue an array of Date objects or array of strings ? 
Also is it possible to follow the other form components by using value and defaultValue ? See input and textarea components https://github.com/GovTechSG/sgds-web-component/blob/master/src/components/Textarea/sgds-textarea.ts

Cos so far the other forms component are using value and @defaultValue decorator to control the value of the component.